src/qml/editorwidgets/CheckBox.qml

```diff
@@ -15,20 +15,16 @@ Item {
 
     property var currentValue: value
     //if the field type is boolean, ignore the configured 'CheckedState' and 'UncheckedState' values and work with true/false always
-    readonly property bool isBool: field.type == 1
+    readonly property bool isBool: field.type == 1 //needs type coercion
 
-    checked: isBool ? value : value === config['CheckedState']
+    checked: isBool ? currentValue !== undefined ? currentValue : false : String(currentValue) === config['CheckedState']
```

Comment:
Maybe it would be better to switch this for some `if else` clauses instead of nested onliners, that slowly gets messy to read ;)

Reply:
Well **if** JavaScript, then "Let's go the whole javascript hog." :slightly_smiling_face: 
But I agree with you and I will change it.